### DISCUSSION TITLE
Axum Integration

### DISCUSSION THREAD

**Discussion by mcmah309** [tokio-rs/axum#2691] [Q&A]
### Summary

Is there a reason that the axum example uses a layer rather than a service? 
Does both enable connecting over websockets or do they use long polling instead?
I see that 0.12.0 added middleware, is this something that could not be accomplished with towers layers before?

### axum version

0.7

**Comment by jplatte** [comment]
It looks like you're in the wrong place, axum version 0.12 doesn't exist and there's many examples, not just one.

**Comment by mcmah309** [comment]
Yup. Wrong tab.. my bad 😅
